### PR TITLE
Enhanced error logging

### LINKED ISSUE #5488: enhance error logging for existing object is not controlled by UID - add GVK, Resource Name etc.
### What problem are you facing?

please provide an indication which resource is not controlled by the correct object. If my compostion has 30 resources, it would be good to have an indicator, which resource is not controlled by the correct UID.

```
cannot compose resources: cannot apply composed resource: existing object is not controlled by UID "941a35fb-16b2-49af-be92-55e34e08cfbe"
```

image

### PR DESCRIPTION
**Pull request by NeerajNagure**
### Description of your changes
Added template name to errApplyComposed so that we can get more info about the resource

Fixes #5488 

I have: 

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added or updated unit tests.
- [ ] ~Added or updated e2e tests.~
- [ ] ~Linked a PR or a [docs tracking issue] to [document this change].~
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR.~

Need help with this checklist? See the [cheat sheet].

[contribution process]: https://github.com/crossplane/crossplane/tree/master/contributing
[docs tracking issue]: https://github.com/crossplane/docs/issues/new
[document this change]: https://docs.crossplane.io/contribute/contribute
[cheat sheet]: https://github.com/crossplane/crossplane/tree/master/contributing#checklist-cheat-sheet

### CODE REVIEW

File: internal/controller/apiextensions/composite/composition_pt.go [hunk 1]
```diff
@@ -279,7 +279,7 @@ func (c *PTComposer) Compose(ctx context.Context, xr *composite.Unstructured, re
 				// run again the composition after some other resource is
 				// created or updated successfully. So, we emit a warning event
 				// and move on.
-				events = append(events, event.Warning(reasonCompose, errors.Wrap(err, errApplyComposed)))
+				events = append(events, event.Warning(reasonCompose, errors.Wrapf(err, errApplyComposed, ptr.Deref(t.Name, fmt.Sprintf("%d", i+1)))))
```
Comment (negz) [inline internal/controller/apiextensions/composite/composition_pt.go:282]: I _think_ we might fall back to the array index in some other messages where `t.Name` is `nil`, but I'm not sure we use `i+1` - we might just use `I`. Can you double check to make sure this will be consistent?

**Review by negz — COMMENTED** [review]
Thank you!

File: internal/controller/apiextensions/composite/composition_pt.go (hunk 1 above)
Comment (negz) [reply internal/controller/apiextensions/composite/composition_pt.go:282]: Thanks for checking!

File: internal/controller/apiextensions/composite/composition_pt.go [hunk 2]
```diff
@@ -45,7 +45,7 @@ import (
 const (
 	errGetComposed   = "cannot get composed resource"
 	errGCComposed    = "cannot garbage collect composed resource"
-	errApplyComposed = "cannot apply composed resource"
+	errApplyComposed = "cannot apply composed resource %q"
```
Comment (negz) [inline internal/controller/apiextensions/composite/composition_pt.go:48]: Can you rename this and move it down to the block below with all the other `errFmt` strings, please.

 
```suggestion
	errFmtApplyComposed = "cannot apply composed resource %q"
```

**Comment by negz** [comment]
@NeerajNagure Apart from renaming to `errFmt...` this looks good to me. Can you please fill out the checklist on the PR description per the linked cheat sheet.

**Comment by NeerajNagure** [comment]
@negz I have made all the necessary changes and fixed unit tests

**Review by negz — APPROVED** [review]
Looks good, thank you!

**Comment by negz** [comment]
Hrm, actually it looks like `make check-diff` is failing. Can you rebase on master and run `make reviewable` again please.

**Comment by NeerajNagure** [comment]
@negz I have merged the latest changes from master branch and also ran make reviewable

**Comment by negz** [comment]
@NeerajNagure It seems like something went wrong with your rebase. 😄 This PR now has 21 unrelated commits.

**Comment by NeerajNagure** [comment]
@negz I have fixed all the issues related to rebase.Please check it and inform me any other changes are supposed to be made

**Review by Pocokwins — APPROVED** [review]

File: apis/apiextensions/fn/proto/v1beta1/run_function.pb.go [hunk 3]
```diff
@@ -15,7 +15,7 @@
 
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.33.0
+// 	protoc-gen-go v1.25.0-devel
```
Comment (jbw976) [inline apis/apiextensions/fn/proto/v1beta1/run_function.pb.go:18]: I wouldn't expect this file to be changing at all, especially for it to be reverting this back to an older version of `protoc-gen-go`.

Any idea why this change is being included? 🤔

Comment (NeerajNagure) [reply apis/apiextensions/fn/proto/v1beta1/run_function.pb.go]: @jbw976 Yea, it happened because my protoc-gen-go version was lower so it modified that latest version in `run_function.pb.go`. I have fixed that issue now

**Review by negz — APPROVED** [review]
Thanks for persisting! This looks great.
